pkg/trace/api/api.go

```diff
@@ -542,32 +578,36 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 		log.Errorf("Cannot decode %s traces payload: %v", v, err)
 		return
 	}
-	traces := dectraces.Traces
-	if !dectraces.RanHook {
+	if !ranHook {
 		// The decoder of this request did not run the pb.MetaHook. The user is either using
 		// a deprecated endpoint or Content-Type, or, a new decoder was implemented and the
 		// the hook was not added.
 		log.Debug("Decoded the request without running pb.MetaHook. If this is a newly implemented endpoint, please make sure to run it!")
 		if _, ok := pb.MetaHook(); ok {
 			log.Warn("Received request on deprecated API endpoint or Content-Type. Performance is degraded. If you think this is an error, please contact support with this message.")
-			runMetaHook(traces)
+			runMetaHook(tp.Chunks)
 		}
 	}
 	if n, ok := r.replyOK(v, w); ok {
 		tags := append(ts.AsTags(), "endpoint:traces_"+string(v))
 		metrics.Histogram("datadog.trace_agent.receiver.rate_response_bytes", float64(n), tags, 1)
 	}
 
-	atomic.AddInt64(&ts.TracesReceived, int64(len(traces)))
+	atomic.AddInt64(&ts.TracesReceived, int64(len(tp.Chunks)))
 	atomic.AddInt64(&ts.TracesBytes, req.Body.(*apiutil.LimitedReader).Count)
 	atomic.AddInt64(&ts.PayloadAccepted, 1)
 
-	cid := req.Header.Get(headerContainerID)
+	containerTags := getContainerTags(tp.ContainerID)
+	if containerTags != "" {
```

Comment:
I'm going to possibly annoy you again with my naming nits, but I think this variable name is too long, given that t's only used a few lines below. We don't need that much context for the reader.
```suggestion
	if ctags := getContainerTags(tp.ContainerID); ctags != "" {
```